src/ORM/Table.php

```diff
@@ -583,7 +584,11 @@ public function setSchema($schema)
                 unset($schema['_constraints']);
             }
 
-            $schema = new TableSchema($this->getTable(), $schema);
+            $connection = $this->getConnection();
+            if ($connection === null) {
```

Comment:
The null check isn't necessary. Throughout the `Table` class there are multiple calls to `$this->getConnection()` which don't check it's return type.

Ensuring that `$this->_connection` is set could be done in `getConnection()` as it's done in 4.x.
https://github.com/cakephp/cakephp/blob/11805ef531f6c7368c93ed0816ea65f6a26cd9f2/src/ORM/Table.php#L479-L488

Reply:
I did this check here because many tests failed with null pointer exception as there are many places where tables are constructed with `new` keyword like `new Table(['schema' => $schema])` with no connection. I have no problem in moving that part to `getConnection` but I am afraid that may break someones code as it would set the connection in the constructor before `initialize` method of table class is invoked where the connection could be overridden.

Reply:
It won't create an issue. Except for tests table instances are created by `TableLocator` which always sets the `connection` key in config passed to `Table::_construct()`.